services:
  postgres:
    container_name: postgres_container
    image: 'postgres:16.3'
    environment:
      - 'POSTGRES_DB=bot_app'
      - 'POSTGRES_PASSWORD=dev'
      - 'POSTGRES_USER=dev'
    ports:
      - "5433:5432"
    volumes:
      -  bot_app_postgres:/var/lib/postgresql/data
    networks:
      - bot_app

  pgadmin:
    container_name: pgadmin_container
    image: dpage/pgadmin4:7.2
    environment:
      PGADMIN_DEFAULT_EMAIL: "habrpguser@habr.com"
      PGADMIN_DEFAULT_PASSWORD: "dev"
      PGADMIN_CONFIG_SERVER_MODE: "False"
    volumes:
      - pgadmin-data:/var/lib/pgadmin
    ports:
      - "5050:80"
    restart: unless-stopped
    deploy:
      resources:
        limits:
          cpus: '0.5'
          memory: 1G
    networks:
      - bot_app

  prometheus:
    image: prom/prometheus
    container_name: prometheus
    volumes:
      - ./prometheus/prometheus.yml:/etc/prometheus/prometheus.yml
    ports:
      - 9090:9090
    networks:
      - bot_app
#    extra_hosts:    # ERROR !!! uncoment on LINUX ?
#      - "host.docker.internal:172.17.0.1"
  grafana:
    image: grafana/grafana:12.0.2
    container_name: grafana_with_mon
    environment:
      - GF_SECURITY_ADMIN_USER=dev
      - GF_SECURITY_ADMIN_PASSWORD=dev
      - GF_USERS_ALLOW_SIGN_UP=false
    ports:
      - 3000:3000
    networks:
      - bot_app
    links:
      - prometheus
    depends_on:
      - prometheus
    volumes:
      - grafana_data_bot:/var/lib/grafana

volumes:
  bot_app_postgres:
  pgadmin-data:
  grafana_data_bot:

networks:
  bot_app:
    driver: bridge